Libplanet/Store/DefaultStore.cs

```diff
@@ -560,16 +579,28 @@ HashDigest<SHA256> blockHash
                     )
                 )
             );
-            using (var stream = new MemoryStream())
+
+            UPath path = StatePath(blockHash);
+            UPath dirPath = path.GetDirectory();
+            CreateDirectoryRecursively(_states, dirPath);
+
+            using (var buffer = new MemoryStream(new Codec().Encode(serialized)))
```

Comment:
As `Bencodex.Codec` has an overload `Encode(IValue, Stream)` as well, you don't have a `buffer` but could directly write to `deflate` or `file` instead, e.g., `new Codec().Encode(serialized, deflate)`.